src/State/Scope.php

```diff
@@ -27,6 +27,11 @@ final class Scope
      */
     private $user;
 
+    /**
+     * @var Context The list of contexts associated to this scope
+     */
+    private $contexts;
```

Comment:
For simplicity let's use an `array<string, array<string, mixed>>` too here. I don't think that there is any good reason to use the `Context` object which we may remove in `3.0` given that `contexts is able to override any data of the interfaces, thus effectively bypassing the validation provided by these objects